Doc/library/asyncio-queue.rst

```diff
@@ -92,6 +97,21 @@ Queue
 
       Return the number of items in the queue.
 
+   .. method:: shutdown(immediate=False)
+
+      Shut down the queue, making :meth:`~Queue.get` and :meth:`~Queue.put`
+      raise :exc:`QueueShutDown`.
+
+      By default, :meth:`~Queue.get` on a shut down queue will only
+      raise once the queue is empty. Set *immediate* to true to make
+      :meth:`~Queue.get` raise immediately instead.
+
+      All blocked callers of :meth:`~Queue.put` will be unblocked. If
+      *immediate* is true, also unblock callers of :meth:`~Queue.get`
+      and :meth:`~Queue.join`.
+
```

Comment:
Sorry but I have a doubt, shouldn't this documentation block be rather:
```     
    All blocked callers of :meth:`~Queue.put` and :meth:`~Queue.get` 
    will be unblocked. If *immediate* is true, also unblock callers of 
    :meth:`~Queue.join`.
```
In event of change,  the docstring of the `shutdown` method must be updated.

Reply:
`join` callers aren't necessarily even unblocked anyway, if consumers are processing any items. I should probably say that a task is marked as done for each item in the queue if immediate shutdown.

Also, I think the threading queue docs are the same.

Reply:
> a task is marked as done for each item in the queue if immediate shutdown.

It's very precise, better.

> Also, I think the threading queue docs are the same.

Yes, I commented here so as not to forget (see https://github.com/python/cpython/pull/117532#issuecomment-2041150494). 

English is your native language, I think it'is best if you update documentations and docstrings

Reply:
I've made a PR: #117621